interpreter/cling/lib/Interpreter/DeclCollector.h

```diff
@@ -121,7 +121,7 @@ namespace cling {
 
     void DeclRead(clang::serialization::DeclID, const clang::Decl *D) {
       assert(D);
-      if (!D->hasOwningModule())
+      if (!D->hasOwningModule() || !D->isUsed() || D->isImplicit() || D->isInStdNamespace() || !D->isFromASTFile())
```

Comment:
How is it possible for a decl to have an owning module but not to come from an ASTFile? Could you make this into an assert? Eg `assert(D->isFromASTFile());`.